### ISSUE TITLE
[SOL-79061] Very verbose logging when connection to broker is interuppted:

### ISSUE THREAD

**Issue by GreenRover** [SolaceProducts/solace-spring-cloud#174]
When the connection between application an broker is interrupted,
i get  thounds of exception per second:

```
2022-09-16 15:59:59,999 WARN [pool-26-thread-1] com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener  Received error while trying to read message from endpoint INTERNAL/tms/capaopt/oda/d-geheim/odainterfaceprocessor/v3/operatingstate/executionura
com.solacesystems.jcsmp.StaleSessionException: Tried to call receive on a stopped message consumer.
      at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.throwClosedException(FlowHandleImpl.java:1900)
      at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:852)
      at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:819)
      at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:279)
      at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:211)
      at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.receive(InboundXMLMessageListener.java:93)
      at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.run(InboundXMLMessageListener.java:73)
      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
      at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (JCSMPTransportException) Error communicating with the router. (KeepAlive)
      at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2320)
      ... 4 common frames omitted
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (Client name: geheim/2711347/#00100002/JOIVd_o7yW   Local addr: 10.1.1.1 Local port: 56744   Remote addr: geheim  Remote port: 15443) - Timeout happened when reading response from the router.
      at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:240)
      at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:97)
      at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:399)
      at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:106)
      at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2156)
      ... 4 common frames omitted
Caused by: java.net.SocketTimeoutException: null
      at java.base/sun.nio.ch.SocketAdaptor$SocketInputStream.read(SocketAdaptor.java:214)
      at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:103)
      at java.base/java.io.InputStream.read(InputStream.java:205)
      at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.readBytes(SSLSmfClient.java:549)
      at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.doPostNoResponse(SSLSmfClient.java:292)
      at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.doPost(SimpleSmfClient.java:230)
      at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:196)
      ... 8 common frames omitted
```

A broken broker connection should not throw exception.
There for a #145 health indicator should report an issue.

Some else also notices this problem.
https://solace.community/discussion/1438/stalesessionexception-spam-when-disconnected-from-host

**Comment by mackenza** [comment]
This has been logged in the Solace Jira and we are targeting Q1CY2023 for a fix.

**Comment by carolmorneau** [comment]
Closed with #215

**Comment by GreenRover** [comment]
Still NOT fixed.

Test environment:
```
solace:
  java:
    host: ${SOLACE_HOSTS}
    msgVpn: ${SOLACE_MSG_VPN}
    clientUsername: ${SOLACE_USERNAME}
    clientPassword: ${SOLACE_PASSWORD}
    clientName: @project.artifactId@_${HOSTNAME}_${random.uuid}
    connectRetries: 5
    reconnectRetries: 3
    connectRetriesPerHost: 2
    reconnectRetryWaitInMillis: 3000
```

Test case:
- connect to a broker
- interrupt the connection to the broker
- check logs while application is reconnecting (Check was OK)
- wait for all reconnect attempts are passed (reconnectRetryWaitInMillis * reconnectRetries * connectRetriesPerHost)
- check logs while application is death (Check was NOT ok)

Logs of application after passing last reconnect attempt:
```
2023-05-05T10:54:24.821+0200 WARN Received error while trying to read message from endpoint #P2P/QTMP/v:developerproductionhfalq83jbhxsolaceprimary0/scst/an/61d3e879-9c2e-416b-a130-f9c07850db5d/plain/receiveTrainPosition-in-0
com.solacesystems.jcsmp.StaleSessionException: Tried to call receive on a stopped message consumer.
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.throwClosedException(FlowHandleImpl.java:2042) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:920) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:887) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:326) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:258) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.receive(InboundXMLMessageListener.java:123) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.run(InboundXMLMessageListener.java:96) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (JCSMPTransportException) Error receiving data from underlying connection.
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2493) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (Client name: pub_sub_receiving_K65710_9a3de0ba-6408-476e-98ae-33ced0aa5e00   Local port: -1   Remote addr: playground.messaging.solace.cloud  Remote port: 55443) - Error communicating with the router.
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:254) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: java.net.ConnectException: Connection timed out: no further information
	at sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishTimedConnect(SocketChannelImpl.java:1148) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.blockingConnect(SocketChannelImpl.java:1183) ~[?:?]
	at sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:98) ~[?:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.open(SimpleSmfClient.java:905) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.ZSmfClient.open(ZSmfClient.java:62) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.open(SSLSmfClient.java:90) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.doPostNoResponse(SSLSmfClient.java:241) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.doPost(SimpleSmfClient.java:235) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:199) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
2023-05-05T10:54:24.821+0200 WARN Received error while trying to read message from endpoint scst/wk/sensor.temperature/plain/sensor/temperature/_/_
com.solacesystems.jcsmp.StaleSessionException: Tried to call receive on a stopped message consumer.
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.throwClosedException(FlowHandleImpl.java:2042) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:920) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:887) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:326) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:258) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.receive(InboundXMLMessageListener.java:123) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.run(InboundXMLMessageListener.java:96) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (JCSMPTransportException) Error receiving data from underlying connection.
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2493) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (Client name: pub_sub_receiving_K65710_9a3de0ba-6408-476e-98ae-33ced0aa5e00   Local port: -1   Remote addr: playground.messaging.solace.cloud  Remote port: 55443) - Error communicating with the router.
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:254) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: java.net.ConnectException: Connection timed out: no further information
	at sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishTimedConnect(SocketChannelImpl.java:1148) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.blockingConnect(SocketChannelImpl.java:1183) ~[?:?]
	at sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:98) ~[?:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.open(SimpleSmfClient.java:905) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.ZSmfClient.open(ZSmfClient.java:62) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.open(SSLSmfClient.java:90) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.doPostNoResponse(SSLSmfClient.java:241) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.doPost(SimpleSmfClient.java:235) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:199) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
2023-05-05T10:54:24.821+0200 WARN Received error while trying to read message from endpoint scst/wk/sensor.temperature/plain/sensor/temperature/_/_
com.solacesystems.jcsmp.StaleSessionException: Tried to call receive on a stopped message consumer.
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.throwClosedException(FlowHandleImpl.java:2042) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:920) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.impl.flow.FlowHandleImpl.receive(FlowHandleImpl.java:887) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:326) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.util.FlowReceiverContainer.receive(FlowReceiverContainer.java:258) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.receive(InboundXMLMessageListener.java:123) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at com.solace.spring.cloud.stream.binder.inbound.InboundXMLMessageListener.run(InboundXMLMessageListener.java:96) ~[spring-cloud-stream-binder-solace-core-3.4.0.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (JCSMPTransportException) Error receiving data from underlying connection.
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2493) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: com.solacesystems.jcsmp.JCSMPTransportException: (Client name: pub_sub_receiving_K65710_9a3de0ba-6408-476e-98ae-33ced0aa5e00   Local port: -1   Remote addr: playground.messaging.solace.cloud  Remote port: 55443) - Error communicating with the router.
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:254) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
Caused by: java.net.ConnectException: Connection timed out: no further information
	at sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishTimedConnect(SocketChannelImpl.java:1148) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.blockingConnect(SocketChannelImpl.java:1183) ~[?:?]
	at sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:98) ~[?:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.open(SimpleSmfClient.java:905) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.ZSmfClient.open(ZSmfClient.java:62) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.open(SSLSmfClient.java:90) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SSLSmfClient.doPostNoResponse(SSLSmfClient.java:241) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.smf.SimpleSmfClient.doPost(SimpleSmfClient.java:235) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpChannel.executePostOnce(TcpChannel.java:199) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.ChannelOpStrategyClient.performOpen(ChannelOpStrategyClient.java:99) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.performOpenSingle(TcpClientChannel.java:426) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel.access$800(TcpClientChannel.java:122) ~[sol-jcsmp-10.16.0.jar:?]
	at com.solacesystems.jcsmp.protocol.impl.TcpClientChannel$ClientChannelReconnect.call(TcpClientChannel.java:2329) ~[sol-jcsmp-10.16.0.jar:?]
	... 4 more
```

Expected:
- Application logs reconnecting attempts
- Application logs last reconnecting attempt is failed
- Health status if DOWN when last reconnect attempt is passed
- Application goes never crazy:
  - Application is not logging > 100 per sec
  - Application is not consuming full CPU core for message reading
